ext/hash/xxhash/xxhash.h

```diff
@@ -2934,6 +2934,7 @@ enum XXH_VECTOR_TYPE /* fake enum */ {
   && defined(__OPTIMIZE__) && !defined(__OPTIMIZE_SIZE__) /* respect -O0 and -Os */
 #  pragma GCC push_options
 #  pragma GCC optimize("-O2")
+#  pragma GCC optimize("no-reorder-blocks-and-partition")
```

Comment:
Are all of these changes actually needed, or only this one (because it overrides with `-O2`)? If it's only this one that's really necessary, I think we can just drop the GCC optimize usage here entirely.

Reply:
Agree. Seems -O2 is not necessary here, we should remove this logic.